File: Codes/2410-maximum-matching-of-players-with-trainers.java
/*
 * @Author: LetMeFly
 * @Date: 2025-07-13 16:44:48
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-07-13 17:06:30
 */
import java.util.Arrays;

class Solution {
    public int matchPlayersAndTrainers(int[] players, int[] trainers) {
        Arrays.sort(players);
        Arrays.sort(trainers);
        int ans = 0;
        for (int p = 0, t = 0; p < players.length && t < trainers.length; t++) {
            if (trainers[t] >= players[p]) {
                ans++;
                p++;
            }
        }
        return ans;
    }
}

File: Codes/2410-maximum-matching-of-players-with-trainers.py
'''
Author: LetMeFly
Date: 2025-07-13 16:44:48
LastEditors: LetMeFly.xyz
LastEditTime: 2025-07-13 17:00:27
'''
from typing import List

class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort()
        trainers.sort()
        ans = p = t = 0
        while t < len(trainers) and p < len(players):
            if trainers[t] >= players[p]:
                ans += 1
                p += 1
            t += 1
        return ans